src/symmetric_crypto.cpp

```diff
@@ -0,0 +1,382 @@
+#include <utility>
+#include <strings.h>
+#include <deque>
+
+
+#include <openssl/evp.h>
+
+#include <mococrw/error.h>
+#include "mococrw/symmetric_crypto.h"
+#include "mococrw/openssl_wrap.h"
+
+namespace mococrw
+{
+
+class AESCipher::Impl
+{
+public:
+    Impl(SymmetricCipherMode mode, SymmetricCipherPadding padding,
+         const std::vector<uint8_t> &secretKey, const std::vector<uint8_t> &iv, Operation operation,
+         std::unique_ptr<CipherMemoryStrategyI> memoryStrategy = std::make_unique<QueueOfVectorsMemoryStrategy>())
```

Comment:
The idea was to let developer to introduce their own memory strategies. On the other hand, we can make it private and expose it later if it would be a neec or when we have more than one strategy in place.

Reply:
I'll move it into a separate files which are not part of the public interface.